src/expr/node_manager.h

```diff
@@ -178,7 +178,7 @@ class NodeManager {
   std::vector<NodeManagerListener*> d_listeners;
 
   /** A list of datatypes owned by this node manager. */
-  std::vector<Datatype*> d_ownedDatatypes;
+  std::vector<std::shared_ptr<DType> > d_ownedDTypes;
```

Comment:
I don't quite see a reason for this to be a `shared_ptr`: We only seem to be returning references to the `DType`s owned by this vector, so no one else can actually increment the reference count of the `shared_ptr`s (so we might as well just use `unique_ptr`s).

Reply:
I'm not sure how I could change this currently.  NodeManager actually doesn't do the creation, ExprManager does (and in fact the DType is a shared_ptr within Datatype).  My understanding is one can't convert shared_ptr to unique_ptr.  Or more concretely, what change are you suggesting that would allow this?

Reply:
We've discussed this quite a bit offline and we've come to the conclusion that it is easiest to leave it as it is now (there is a shared ownership between `Datatype::d_internal` and `NodeManager::d_ownedDTypes`). Ideally, registered `DType`s would only be owned by the `NodeManager` but this will require some additional refactoring that is out-of-scope.